diff --git a/src/twisted/mail/smtp.py b/src/twisted/mail/smtp.py
index 7725ce588..9b5cdb293 100644
--- a/src/twisted/mail/smtp.py
+++ b/src/twisted/mail/smtp.py
@@ -2035,7 +2035,6 @@ class ESMTPSender(SenderMixin, ESMTPClient):
         else:
             try:
                 context = ssl.ClientContextFactory()
-                context.method = ssl.SSL.TLSv1_METHOD
                 return context
             except AttributeError:
                 return None
